### DISCUSSION TITLE
Typing a promise service

### DISCUSSION THREAD

**Discussion by geekox86** [statelyai/xstate#3572] [Q&A]
Greetings,

I have followed the documentation on typing a promise service, however I am getting this in my typegen file:
```
"done.invoke.someService": {
    type: "done.invoke.someService";
    data: unknown;
    __tip: "See the XState TS docs to learn how to strongly type this.";
};
```

Here is my services type:
```
type ServiceEvents<TData> = {
  someService: { data: TData };
};

// also I have this line in my schema where `TData` is coming from the function that creates the machine:
services: {} as ServiceEvents<TData>
```

Also, is it possible to type the errors coming from the promise service?

Any help is appreciated.

**Comment by Andarist** [comment]
Please always try to share a repro case in a runnable form - either by providing a git repository to clone or a codesandbox.
